compiler/passes/codegen.cpp

```diff
@@ -765,7 +780,7 @@ static void protectNameFromC(Symbol* sym) {
 
 
 // TODO: Split this into a number of smaller routines.<hilde>
-static void codegen_header() {
+static void codegen_header(bool isHeader) {
```

Comment:
Ah, I see by the alterations to the body of this function that there are times when it will be called with both modes.  The actions in both cases seems to be fairly separate, though - perhaps this function should be split in two, with one handling the true case and one handling the false case?  The false function could be renamed to refer entirely to the new .c file you created.